assessment/libs/stats.php

```diff
@@ -2196,4 +2197,393 @@ function dotplot($a,$label,$dotspace=1,$labelspace=null,$width=300,$height=150)
+//  for twoway WITH replication - example: $arr=[[[4,5,6,5],[7,9,8,12],[10,12,11,9]],[[6,6,4,4],[13,15,12,12],[12,13,10,13]]]
+//  for twoway WITHOUT replication - example: $arr=[[53,61,51],[47,55,51],[46,52,49],[50,58,54],[49,54,50]] 
+// replication: Optional - boolean (true or false) it specifies whether the ANOVA with replication
+//             (multiple observations for each group) or without replication (one observation per group)
+//             is to be performed. The default is False - without replication.
+//  
+// Returns:
+// F ratio and the corresponding P value for Factor A, Factor B and their Interaction (if replication is true)
+// as an array in the form  array([F_A,P_A],[F_B,P_B],[F_I,P_I]). 	
+
+function anova2way_f($arr, $replication=False){
+	$k=anova2way($arr,$replication);
+	if($replication==True){
+		$ans=[[$k[0][3],$k[0][4]],[$k[1][3],$k[1][4]],[$k[2][3],$k[2][4]]];
+	}
+		else{
+			$ans=[[$k[0][3],$k[0][4]],[$k[1][3],$k[1][4]]];
+	}
+	
+	return($ans);
+}
+
+
+
+//-----------------------------------------------ANOVA Table---------------------------------------------
+// Function: anova_table(arr,[factor=1, replication=False, roundto=12, nameA="factorA", nameB="factorB "])
+// Returns ANOVA table for both oneway and twoway ANOVA - display only. The output of anova1way_arr() and 
+// anova2way_arr() can be used as the input array for this function.
+//
+// Parameters:
+// arr: An array in the follwing form: 
+//   for oneway: [[SS_A, df_A, MS_A, F_A, P_A],[SS_E,df_E,MS_E],[SS_T,df_T]]
+//   for twoway WITHOUT replication: [[SS_A, df_A, MS_A, F_A, P_A],[SS_B, df_B, MS_B, F_B, P_B],[SS_E,df_E,MS_E],[SS_T,df_T]]
+//   for twoway WITH replication: [[SS_A, df_A, MS_A, F_A, P_A],[SS_B, df_B, MS_B, F_B, P_B],[SS_I, df_I, MS_I, F_I, P_I],[SS_E,df_E,MS_E],[SS_T,df_T]]
+// factor: number of factors considered in ANOVA - 1 for one-way and 2 for two-way. The default is 1, one-way ANOVA.
+// replication: Optional - boolean (true or false) it specifies whether the ANOVA tewoway with replication
+//             (multiple observations for each group) or without replication (one observation per group)
+//             is performed. The default is False - without replication.
+// roundto: Optional - number of decimal places to which data should be rounded off; 
+//          default is 12 decimal places. 
+// NameA: Optional - the name of factor A as string to be displayed in the table. Default is "Factor A".
+// NameB: Optional - the name of factor B as string to be displayed in the table. Default is "Factor B".
+// Returns:
+// ANOVA table for displaying data. 
+
+
+function anova_table(array $array, int $factor = 1, $rep=False, int $roundto=12, string $f1="Factor A", string $f2="Factor B"){
+	if ($factor!=1 && $factor!=2) { echo 'error: the factor variable only expects 1 for one-way and 2 for two-way ANOVA'; return '';}
+	if (!function_exists('calconarray')) {
+        require_once(__DIR__.'/assessment/macros.php');

```

Comment:
This shouldn't be necessary; macros.php is always loaded when the macro libraries are used.